docs/proposals/20230123-azure-service-operator.md

```diff
@@ -0,0 +1,147 @@
+---
+title: Managing Azure Resources with Azure Service Operator
+authors:
+  - "@nojnhuh"
+reviewers:
+  - "@CecileRobertMichon"
+  - "@jackfrancis"
+  - "@mattchr"
+creation-date: 2023-01-23
+last-updated: 2023-01-23
+status: provisional
+see-also:
+  - https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/416
+replaces:
+  - "/docs/proposals/20210716-async-azure-resource-creation-deletion.md"
+---
+
+# Managing Azure Resources with Azure Service Operator
+
+## Table of Contents
+
+- [Managing Azure Resources with Azure Service Operator](#managing-azure-resources-with-azure-service-operator)
+  - [Table of Contents](#table-of-contents)
+  - [Glossary](#glossary)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals/Future Work](#non-goalsfuture-work)
+  - [Proposal](#proposal)
+    - [User Stories](#user-stories)
+      - [Story 1](#story-1)
+    - [Requirements (Optional)](#requirements-optional)
+      - [Functional Requirements](#functional-requirements)
+        - [FR1](#fr1)
+      - [Non-Functional Requirements](#non-functional-requirements)
+        - [NFR1](#nfr1)
+    - [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+    - [Security Model](#security-model)
+    - [Risks and Mitigations](#risks-and-mitigations)
+  - [Alternatives](#alternatives)
+  - [Upgrade Strategy](#upgrade-strategy)
+  - [Additional Details](#additional-details)
+    - [Test Plan](#test-plan-optional)
+    - [Graduation Criteria](#graduation-criteria-optional)
+  - [Implementation History](#implementation-history)
+
+## Glossary
+
+- **Azure Service Operator (ASO)** - ([link](https://azure.github.io/azure-service-operator/)) a collection of Kubernetes CRDs and controllers which extend the Kubernetes API to manage Azure resources like virtual machines and resource groups.
+
+## Summary
+
+This proposal describes how CAPZ could utilize Azure Service Operator (ASO) to create, read, update, and delete Azure resources with a Kubernetes-native interface.
+
+## Motivation
+
+CAPZ currently interfaces with the Azure APIs using https://github.com/Azure/azure-sdk-for-go, a thin wrapper over Azure's REST API, to manage Azure resources backing Cluster API clusters. The Azure Go SDK has recently deprecated its set of APIs that CAPZ uses in favor of a new set of backwards-incompatible APIs. Because the deprecated Azure Go SDK APIs are no longer receiving updates to adopt new Azure API versions, CAPZ must adopt the new APIs to be able to expose the newest features in Azure to Cluster API users.
+
+Because ASO already uses the new APIs, adopting ASO in CAPZ accomplishes the goal of migrating off of the deprecated APIs. In addition, the declarative, Kubernetes-native interface exposed by ASO unlocks opportunities to simplify how CAPZ reconciles resources in Azure.
+
+### Goals
+
+- Adopt ASO in CAPZ to handle creating, updating, and deleting Azure resources
```

Comment:
I don't think a goal should be to "adopt ASO" since that's the "how" not the "what" 

A better goal might be

> Separate concerns for cloud resource reconciliation from the transformation of Cluster API custom resource specs to cloud resource specs

(took from #416)